android/beagle/src/main/java/br/com/zup/beagle/android/context/operations/exception/ExceptionFactory.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.context.operations.exception
+
+import br.com.zup.beagle.android.context.operations.exception.strategy.ExceptionOperationTypes
+import br.com.zup.beagle.android.context.operations.exception.strategy.ExceptionParameterTypes
+import br.com.zup.beagle.android.context.operations.exception.strategy.ExceptionTypes
+import br.com.zup.beagle.android.context.operations.operation.Operation
+import br.com.zup.beagle.android.context.operations.strategy.string.StringOperationTypes
+
+object ExceptionFactory {
+
+    fun createException (exceptionTypes: ExceptionTypes,
```

Comment:
this method it is really big, what do you think about centralizing the messages in someplace and follow some pattern to get the correct message?
Example:
```suggestion
    fun createException (exceptionTypes: ExceptionTypes, operation: Operation) {
       return EnumMessage.getEnum(exception, operation).getMessageFormatted()
    }
    
   enum EnumMessage(val message: String) {
    NotFound(message = "Could not resolve your operation. Did you check if the Operation is valid?:: $details")
    
fun getEnum(exception, operation): MessageEnum {

//LOGIC  TO GET CORRECT ENUM
    }
    
    fun getMessageFormatted(vararg parameters): {
return message.replace(parameters)// something like that to parameters you want to change similar has in string.xml when you want to add some parameter
    }
   }
    
```

Reply:
Done